apps/dcos_net/src/dcos_net_mesos_listener.erl

```diff
@@ -454,6 +456,10 @@ handle_task_name(TaskObj, _Task) ->
     Name = mget(<<"name">>, TaskObj, undefined),
     mget([<<"discovery">>, <<"name">>], TaskObj, Name).
 
+-spec(handle_task_runtime(jiffy:object(), task()) -> binary() | undefined).
+handle_task_runtime(TaskObj, _Task) ->
+    mget([<<"container">>, <<"type">>], TaskObj, undefined).
```

Comment:
If the function returns undefined, the field is optional and this code [1] can throw `badkey` exception. I'd like to make `runtime` field required.

[1] https://github.com/dcos/dcos-net/pull/102/files#diff-b85318f2b911cad717b851556fb2153dR127

Reply:
```
Default = maps:get(runtime, Task, unknown),
try mget([<<"container">>, <<"type">>], TaskObj) of
    <<"MESOS">> ->
        mesos;
    <<"DOCKER">> ->
        docker;
    Type ->
        lager:warning("", [TaskId, Type]),
        unknown
catch error:{badkey, _} ->
    Default
end
```